controllers/datafoundation.go

```diff
@@ -110,6 +121,16 @@ func dfIsReadyToBeRemoved(r *ManagedFusionOfferingReconciler, offering *v1alpha1
 }
 
 func dfReconcile(offeringReconciler *ManagedFusionOfferingReconciler, offering *v1alpha1.ManagedFusionOffering) (ctrl.Result, error) {
+
+	if !offeringReconciler.AvailableCRDs[StorageClusterCRD] {
+		// we only check for storagecluster CRD because other crds are installed as a dependency via ocs-operator
+		crd := ocsv1.StorageCluster{}
+		err := offeringReconciler.get(&crd)
+		if err != nil && !errors.IsNotFound(err) {
+			return ctrl.Result{}, &utils.ErrAgentRestart{Msg: "restart agent to register watchers for the CRDs"}
+		}
```

Comment:
The logic seems broken here. You are restarting if there is a get error that is not a NotFound error.
The correct logic is: Get the CRD, if the get operation results in any kind of error then request via error. 
If the Get operation was successful then restart